src/qfusion.cpp

```diff
@@ -206,7 +213,7 @@ void QFusion::ApplyControlledSingleBit(
     // We record that this bit is controlled by the bits in its control list.
     if (bitBuffers[target] == NULL) {
         for (bitLenInt i = 0; i < controlLen; i++) {
-            bitControls[controls[i]].push_back(target);
+            bitControls[controls[i]].insert(target);
```

Comment:
Switching to set away from vector imposes a sort penalty on each insertion.  If the sequence is many-additions followed by an operation, it can often be more efficient to perform the additions in a vector, then apply std::sort once prior to the operation.